verify.go

```diff
@@ -54,6 +54,21 @@ func (p7 *PKCS7) VerifyWithChainAtTime(truststore *x509.CertPool, currentTime ti
 	return nil
 }
 
+// SigningTimeNotValidError is returned when signing time attribute falls
+// outside of the signer certificate validity.
+type SigningTimeNotValidError struct {
+	SigningTime time.Time
+	NotBefore   time.Time // NotBefore of signer
+	NotAfter    time.Time // NotAfter of signer
```

Comment:
I'm not sure it makes sense to include these because the caller of p7.Verify() will also have access to the signer cert which can directly access the NotBefore/NotAfter. On the other hand it's nice and explicit to have the actual checked values right here in the error, too.